lib/environment.js

```diff
@@ -56,10 +57,21 @@ class NunjucksEnvironment extends nunjucks.Environment {
       const filterPath = path.join(unit.path, 'app/extend/filter.js');
       const filters = this.app.loader.loadFile(filterPath) || {};
       for (const name of Object.keys(filters)) {
-        this.addFilter(name, filters[name]);
+        this.addFilter(name, this[FILTER_WRAPPER](filters[name]), true);
       }
     }
   }
+
+  [FILTER_WRAPPER](filterFunc) {
+    return (...args) => {
+      const callback = args.pop();
+      // use Promise.resolve instead co, when support async/await
+      co(filterFunc(...args)).then(
```

Comment:
https://github.com/eggjs/egg-core/blob/master/lib/loader/mixin/controller.js#L72

try this to support both generator function and async function